### PR TITLE
[SYCL] Add value_type alias to sycl::vec class

### PR DESCRIPTION
**Pull request by lbushi25**
This PR adds the value_type alias for the data type of a sycl::vec object as described in the spec.

### CODE REVIEW

**Comment by bader** [comment]
Please, add a regression test.
@intel/sycl-language-enabling-triage, I suppose all SYCL core spec features missing in the DPC++ implementation should go with SYCL-CTS patch. Right?

**Comment by lbushi25** [comment]
> Please, add a regression test. @intel/sycl-language-enabling-triage, I suppose all SYCL core spec features missing in the DPC++ implementation should go with SYCL-CTS patch. Right?

Done

**Comment by lbushi25** [comment]
I also had to make a change to the file generic_type_traits.hpp at line 521. There were two declarations of function template check that would result in identical instantiations so I added an enable clause to disable one of them in the case that it is instantiated with sycl::vec.

File: sycl/test/regression/vec_value_type.cpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// RUN: %clangxx -fsycl %s
```
Comment (bader) [inline sycl/test/regression/vec_value_type.cpp:1]:
```suggestion
// RUN: %clangxx -fsycl -fsyntax-only %s
```
from https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#dpc-headers-and-runtime-tests
> if you only need to check that compilation succeeds, please use [-fsyntax-only](https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fsyntax-only) compiler flag for such tests: it instructs the compiler to launch reduced set of commands and produce no output (no need to add -o %t.out).

File: sycl/test/regression/vec_value_type.cpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+#include <sycl/sycl.hpp>
+
+using namespace std;
```
Comment (bader) [inline sycl/test/regression/vec_value_type.cpp:4]: std include are missing (e.g. `#include <type_traits>`)

Comment (lbushi25) [reply sycl/test/regression/vec_value_type.cpp:4]: They must be included under sycl.hpp as this compiles fine locally.

Comment (bader) [reply sycl/test/regression/vec_value_type.cpp:4]: You use `std::is_same` and SYCL spec doesn't guarantee that sycl.hpp must provide this template. 
You must explicitly include headers with declarations you use and do not assume that DPC++ implementation will always implicitly include `<type_traits>` header, even though it does so today.

**Review by bader — APPROVED** [review]

**Comment by lbushi25** [comment]
@intel/llvm-gatekeepers ping.

**Pull request merged**